docs/deployment/microbit-deployment-draft.md

```diff
@@ -0,0 +1,166 @@
+
+Not only could this enable direct viewing of dependent source (including the underlying C/C++), it could offer in some editors a list of known bugs in that version, and url links to the version that fixed it.
+
+### Semantic versioning
+
+The editor will use semantic versioning (http://semver.org/). In addition to breaking API changes, the following things will be considered ‘MAJOR’ changes
+1) Addition of significant features if they interact with the existing deployed features
+2) Things that would change screenshots of the editor that are published online or in books.
+
+Additional buttons on a toolbar are fine, but removing or renaming existing buttons should be considered a breaking change.
+
+Anything that could stop a script that previously worked from working must be considered a breaking change. In the case of MicroPython this means that things increasing memory consumption or reducing the flash available for a users’ script are likely breaking changes.
+
+{<< what other things should constitute a breaking change, and should we allow certain kinds of breaking change inside a version of the editor? IE can we support more than one MicroPython version for a particular deployed version of the front-end?>>}}
+
+It may be desirable to agree a high water mark in memory and growing beyond this is considered a breaking change. It would then be necessary to police memory growth but people do already do this I think. This would allow small increases in memory usage to be possible without a breaking change.
+
+### URL versioning scheme
+
+For each published major version, the deployed editor will live at
+
+`python.microbit.org/v<major version>/editor.html`
+
+Because the use of semantic versioning ensures that things will not break for the user when upgrading minor versions, there will not be fine-grained control over which minor version a user gets. 
+
+For example visiting `python.microbit.org/v2/` will serve version `2.<latest minor>`
+
+### Versions in URLs
+
+When a script is published or shared, it must be published including the major version of the editor (and therefore the MicroPython stack) that it was created with so that anyone following the link will be taken back to the version the script was created with.
+
+If a user visits python.microbit.org/latest/ and is working on the editor there, the URL published should still be versioned, such as python.microbit.org/v1/. 
+
+See “workflows” for discussion of how the user will be notified about newer versions
+
+
+### Versioning in hex files
+
+When the user is provided a hex file, the full semantic editor version must also be embedded into the hex file at a known location so that when unpacking the hex, any version of the editor can offer to load the version that the script was created in.
```

Comment:
I think if we have 1:1 mapping of MicroPython version to editor version that does work BUT, what happens if we start to do things like PXT packages where we have in-editor assembly of modules - this isn't out of the question and I think means we could do with a way to say "this editor version created this hex file". Also, it means that third parties (say Mu, MS) won't have to go and look up the MicroPython version that maps to an editor version in order to redirect a user correctly. (IE if I drop a hex file from PythonEditor into PXT, PXT should be able to redirect me to the correct location!)

One way to do this would be to maintain a mapping of MP  version to editor version (can be many:1) in the 'latest' version, so if you open a hex in /latest?mpversion=ABC it can redirect you to the right editor version?